libs/rtefsutils/src/RteFsUtils.cpp

```diff
@@ -299,22 +301,19 @@ bool RteFsUtils::SetTreeReadOnly(const string& path, bool bReadOnly) {
     return false;
   }
 
-  // Clear base dir restrictions
-  if (!SetFileReadOnly(path, false)) {
+  // Set permission on directory
+  if (!SetFileReadOnly(path, bReadOnly)) {
     return false;
   }
-  // Iterate over file tree
+
+  // Iterate over tree and change permission of files and directories
   for(auto& p: fs::recursive_directory_iterator(path, ec)) {
-    if (fs::is_regular_file(p.path(), ec)) {
+    if (fs::is_regular_file(p.path(), ec) || fs::is_directory(p.path(), ec)) {
```

Comment:
I'm a bit uncertain if we should limit to files or directories here or if we should update any other kind of resource too.
What's your opinion?

Reply:
Well, I think that could be debated. What type of paths do we expect? I'd think files, directories and symlinks in the first place. Lets turn the question around: What could go wrong if we remove the condition?

Reply:
I don't think you can change permission on a symlink, but I'm not sure if it's silent or if an error is raised.
For any other type of resource, I think you can change the permission on a Linux FS.

Reply:
I am sure you can. A symlink is kinda special, as you can either change the symlink itself or the element it links to. According to docs we need to add `fs::filesystem::perm_options::nofollow` in order to affect the symlink and not its target,